### PR TITLE
Expand data.aws_workspaces_bundle With Owner/Name Filtering

### LINKED ISSUE #12496: Expand functionality of data.aws_workspaces_bundle to do local filtering
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Currently, the data lookup for aws_workspaces_bundle only allows for lookup by bundle-id. However, as bundle-id is almost always the attribute required (by @Tensho's work in #11608 for instance) it would be way more helpful to have a lookup that could filter on owner/name/maybe more. 

As it stands, the sdk only allows [filtering by owner](https://docs.aws.amazon.com/sdk-for-go/api/service/workspaces/#DescribeWorkspaceBundlesInput) and further filtering would need to be done locally, post call to the aws api. 

### New or Affected Resource(s)

* data.aws_workspaces_bundle

### Potential Terraform Configuration

```hcl
data "aws_workspaces_bundle" "main" { 
  owner = "AMAZON"
  name  = "Value with Windows 10 and Office 2016"
}
```

### References

* Will help with #11608

### PR DESCRIPTION
**Pull request by Tensho**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12496

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsWorkspaceBundle'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAwsWorkspaceBundle -timeout 120m
=== RUN   TestAccDataSourceAwsWorkspaceBundle_basic
=== PAUSE TestAccDataSourceAwsWorkspaceBundle_basic
=== RUN   TestAccDataSourceAwsWorkspaceBundle_byOwnerName
=== PAUSE TestAccDataSourceAwsWorkspaceBundle_byOwnerName
=== RUN   TestAccDataSourceAwsWorkspaceBundle_bundleIDAndNameConflict
=== PAUSE TestAccDataSourceAwsWorkspaceBundle_bundleIDAndNameConflict
=== CONT  TestAccDataSourceAwsWorkspaceBundle_basic
=== CONT  TestAccDataSourceAwsWorkspaceBundle_bundleIDAndNameConflict
=== CONT  TestAccDataSourceAwsWorkspaceBundle_byOwnerName
--- PASS: TestAccDataSourceAwsWorkspaceBundle_bundleIDAndNameConflict (3.01s)
--- PASS: TestAccDataSourceAwsWorkspaceBundle_basic (21.38s)
--- PASS: TestAccDataSourceAwsWorkspaceBundle_byOwnerName (23.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	24.821s
```

### CODE REVIEW

**Comment by Tensho** [comment]
@bflad @gdavison @ewbankkit Could we merge this?

**Comment by Tensho** [comment]
@bflad @gdavison @ewbankkit Could we merge this?

File: aws/data_source_aws_workspaces_bundle.go [hunk 1]
```diff
@@ -72,25 +75,56 @@ func dataSourceAwsWorkspaceBundle() *schema.Resource {
...
+		}
+
+		if len(resp.Bundles) != 1 {
+			return fmt.Errorf("expect 1 result for workspace bundle %q, got %d", bundleID, len(resp.Bundles))
+		}
+
+		bundle = resp.Bundles[0]
+
+		if bundle == nil {
+			return fmt.Errorf("not found workspace bundle with ID %q", bundleID)
```
Comment (gdavison) [inline aws/data_source_aws_workspaces_bundle.go:95]: Please rephrase to `"no Workspace bundle with ID %q found"`

File: aws/data_source_aws_workspaces_bundle.go [hunk 2]
```diff
@@ -72,25 +75,56 @@ func dataSourceAwsWorkspaceBundle() *schema.Resource {
...
+			}
+
+			return !lastPage
+		})
+		if err != nil {
+			return err
+		}
+
+		if bundle == nil {
+			return fmt.Errorf("not found workspace bundle with name %q", name)
```
Comment (gdavison) [inline aws/data_source_aws_workspaces_bundle.go:119]: Please rephrase to `"no Workspace bundle with name %q found"`

File: aws/data_source_aws_workspaces_bundle.go [hunk 3]
```diff
@@ -72,25 +75,56 @@ func dataSourceAwsWorkspaceBundle() *schema.Resource {
...
+	if bundleID, ok := d.GetOk("bundle_id"); ok {
+		resp, err := conn.DescribeWorkspaceBundles(&workspaces.DescribeWorkspaceBundlesInput{
+			BundleIds: []*string{aws.String(bundleID.(string))},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Bundles) != 1 {
+			return fmt.Errorf("expect 1 result for workspace bundle %q, got %d", bundleID, len(resp.Bundles))
```
Comment (gdavison) [inline aws/data_source_aws_workspaces_bundle.go:89]: Please rephrase to `"expected 1 result for Workspace bundle %q, found %d"`

File: aws/data_source_aws_workspaces_bundle_test.go [hunk 4]
```diff
@@ -33,10 +34,68 @@ func TestAccDataSourceAwsWorkspaceBundle_basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(dataSourceName, "root_storage.0.capacity", "80"),
+					resource.TestCheckResourceAttr(dataSourceName, "user_storage.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "user_storage.0.capacity", "10"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsWorkspaceBundle_bundleIDAndNameConflict(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Tensho) [reply aws/data_source_aws_workspaces_bundle_test.go:63]: Domo arigato 🙇

**Comment by Tensho** [comment]
@gdavison Added fixes according to your review 🚀

**Pull request merged**